libnmstate/nm/sriov.py

```diff
@@ -72,8 +140,65 @@ def get_info(device):
     )
 
     if sriov_setting:
-        info[Ethernet.SRIOV_SUBTREE] = {
-            Ethernet.SRIOV.TOTAL_VFS: sriov_setting.props.total_vfs
-        }
+        sriov_config = {}
+        ifname = device.get_iface()
+        with open(f"/sys/class/net/{ifname}/device/sriov_numvfs", "r") as f:
+            sriov_config[Ethernet.SRIOV.TOTAL_VFS] = int(f.read())
+
+        vfs_config = _create_sriov_vfs_config(sriov_setting)
+        sriov_config[Ethernet.SRIOV.VFS_SUBTREE] = vfs_config
+
+        info[Ethernet.SRIOV_SUBTREE] = sriov_config
 
     return info
+
+
+def _create_sriov_vfs_config(sriov_setting):
+    vfs_config = []
+    vfs_setting = sriov_setting.props.vfs
+    for vf in vfs_setting:
+        vf_config = {}
+        vf_config[Ethernet.SRIOV.VFS.ID] = vf.get_index()
+        if vf.get_attribute(nmclient.NM.SRIOV_VF_ATTRIBUTE_MAC) is not None:
+            vf_config[
+                Ethernet.SRIOV.VFS.MAC
+            ] = nmclient.GLib.Variant.get_string(
+                vf.get_attribute(nmclient.NM.SRIOV_VF_ATTRIBUTE_MAC)
+            )
+        if vf.get_attribute(nmclient.NM.SRIOV_VF_ATTRIBUTE_TRUST) is not None:
+            vf_config[
+                Ethernet.SRIOV.VFS.TRUST
+            ] = nmclient.GLib.Variant.get_boolean(
+                vf.get_attribute(nmclient.NM.SRIOV_VF_ATTRIBUTE_TRUST)
+            )
+        if (
+            vf.get_attribute(nmclient.NM.SRIOV_VF_ATTRIBUTE_MAX_TX_RATE)
+            is not None
+        ):
```

Comment:
These duplicated codes be turn into a function:

```
def _fill_vf_atrribute(vf_cfg, vf, key_name, nm_name, get_func):
    nm_val = vf.get_attribute(nm_name)
    if nm_val is not None:
        cf_cfg[key_name] = get_func(nm_val)
```

Reply:
Yes that is fine but I would need to create 3 different functions as we need to convert the GLib.Variant types to string, int or bool, this is why I am using `nmclient.GLib.Variant.get_uint32` or `nmclient.GLib.Variant.get_boolean`. If you are fine with that I can create them. Also, it would be fine to have a module to handle with GLib.Variant types and use it in other implementations.

Reply:
Done!